src/v/transform/logging/log_manager.cc

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/logging/log_manager.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "base/vassert.h"
+#include "config/configuration.h"
+#include "model/namespace.h"
+#include "random/simple_time_jitter.h"
+#include "strings/utf8.h"
+#include "transform/logging/io.h"
+#include "transform/logging/logger.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/manual_clock.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/coroutine/as_future.hh>
+
+namespace transform::logging {
+namespace {
+using namespace std::chrono_literals;
+} // namespace
+
+namespace detail {
+
+template<typename ClockType>
+class flusher {
+    static constexpr ClockType::duration jitter_amt{50ms};
+    static constexpr int MAX_CONCURRENCY = 10;
+
+public:
+    flusher() = delete;
+    explicit flusher(
+      ss::abort_source* as,
+      transform::logging::client* c,
+      config::binding<std::chrono::milliseconds> fi)
+      : _as(as)
+      , _client(c)
+      , _interval_ms(std::move(fi))
+      , _jitter(_interval_ms(), jitter_amt) {
+        _interval_ms.watch([this]() {
+            _jitter = simple_time_jitter<ClockType>{_interval_ms(), jitter_amt};
+        });
+    }
+
+    template<typename BuffersT>
```

Comment:
Why is `BuffersT` templated?

Reply:
To limit the scope of `buffer_entry`, buffer type, map type. I want to keep the details internal to `manager` and `flusher` out of the header. Maybe there's an angle I'm not seeing?